api/src/main/java/org/openmrs/Allergy.java

```diff
@@ -15,26 +15,60 @@
 import java.util.List;
 import java.util.UUID;
 
+import javax.persistence.AttributeOverride;
+import javax.persistence.AttributeOverrides;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderBy;
+import javax.persistence.Table;
+
 import org.apache.commons.lang3.StringUtils;
+import org.hibernate.annotations.LazyCollection;
+import org.hibernate.annotations.LazyCollectionOption;
 import org.openmrs.util.OpenmrsUtil;
 
 /**
  * Represent allergy
  */
+@Entity
+@Table(name = "allergy")
 public class Allergy extends BaseChangeableOpenmrsData {
 	
 	public static final long serialVersionUID = 1;
 	
+	@Id
+	@GeneratedValue
+	@Column(name = "allergy_id")
 	private Integer allergyId;
 	
+	@ManyToOne(optional = false)
+	@JoinColumn(name = "patient_id")
 	private Patient patient;
 	
+	@Embedded
+	@AttributeOverrides({ @AttributeOverride(name = "allergenType", column = @Column(name = "allergen_type")),
```

Comment:
These are redundant since you are not actually overriding the column names

Reply:
@wluyima should i remove them?

Reply:
@wluyima coz am seeing according to the hbm file the allergenType is part of the allergen component...am abit confused on this

Reply:
I'm saying only remove the @AtttributeOverrides annotation, keep @Embedded only